pkg/lib/util/util.go

```diff
@@ -88,9 +91,70 @@ func Xhtml2Json(rawXhtml []byte) ([]byte, error) {
 		return []byte{}, errors.Wrap(err, "Could not parse xhtml")
 	}
 
-	json, err := mv.JsonIndent("", " ")
+	jsonStr, err := mv.JsonIndent("", " ")
 	if err != nil {
 		return []byte{}, errors.Wrap(err, "Could not convert to json")
 	}
-	return json, nil
+	return jsonStr, nil
+}
+
+// Html2Json parses the raw html data and converts it into a json string
+func Html2Json(rawHtml []byte) ([]byte, error) {
+	var (
+		htmlDoc *goquery.Document
+		err     error
+	)
+
+	htmlDoc, err = goquery.NewDocumentFromReader(bytes.NewReader(rawHtml))
+	if err != nil {
+		return []byte{}, errors.Wrap(err, "Could not parse html")
+	}
+
+	htmlData := parseHtmlNode(htmlDoc.Selection)
+	jsonStr, err := json.MarshalIndent(htmlData, "", " ")
+	if err != nil {
+		return []byte{}, errors.Wrap(err, "Could not convert html to json")
+	}
+
+	return jsonStr, nil
+}
+
+func parseHtmlNode(node *goquery.Selection) map[string]interface{} {
+	tagName := node.Get(0).Data
+	tagData := map[string]interface{}{}
```

Comment:
Use `map[string]any{}`, it's an alias for interface.

Reply:
soll ich das generell an allen stellen im apitesttool anpassen?

Reply:
wenn du magst... aber mir würde es für neue patches reichen